File: AllElementsInTwoBinarySearchTrees.py
# https://leetcode.com/problems/all-elements-in-two-binary-search-trees/
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorder(self,l,node):
        # check if node is a leaf node or not
        if not node:
            return 
        else:
            self.inorder(l,node.left)
            l.append(node.val)
            self.inorder(l,node.right)
        return l
    def getAllElements(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> List[int]:
        lst=[]
        self.inorder(lst,root1)
        self.inorder(lst,root2)
        return sorted(lst)

        

File: AllElementsInTwoBinarySearchTrees.java
// https://leetcode.com/problems/all-elements-in-two-binary-search-trees/
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import javax.swing.tree.TreeNode;

public class AllElementsInTwoBinarySearchTrees {
   public void inorder(List<Integer> l,TreeNode node){
        if(node==null){
            return;
        }
        else{
            inorder(l,node.left);
            l.add(node.val);
            inorder(l,node.right);
        }
        // return l;

    }
    public List<Integer> getAllElements(TreeNode root1, TreeNode root2) {
        List<Integer> lst = new ArrayList<>();
        inorder(lst,root1);
        inorder(lst,root2);
        Collections.sort(lst);
        return lst;
        
    } 
}
